cpp/src/strings/replace/backref_re_medium.cu

```diff
@@ -24,17 +24,17 @@ namespace cudf {
 namespace strings {
 namespace detail {
 
-//
 children_pair replace_with_backrefs_medium(column_device_view const& d_strings,
                                            reprog_device& d_prog,
                                            string_view const& d_repl_template,
-                                           rmm::device_vector<backref_type>& backrefs,
+                                           device_span<backref_type> backrefs,
                                            size_type null_count,
                                            rmm::cuda_stream_view stream,
                                            rmm::mr::device_memory_resource* mr)
 {
+  using Iterator = decltype(backrefs.begin());
   return make_strings_children(
-    backrefs_fn<RX_STACK_MEDIUM>{
+    backrefs_fn<Iterator, RX_STACK_MEDIUM>{
```

Comment:
Can't the iterator type be inferred? Is this just because of the second template param? What if you reverse the order of the template parameters?